Documentation/test-targets-design.md

```diff
@@ -0,0 +1,87 @@
+# Test Target Design
+
+This document describes the design of the BuildTools test targets. For documentation on their usage, see [Test Targets](test-targets-usage.md).
+
+The primary entry points are the [`Test` target](https://github.com/dotnet/buildtools/blob/73375848b73a586a8ac439a1263b605901414ad8/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L336) in Microsoft.DotNet.Build.Tasks and the [`CloudBuild` target](https://github.com/dotnet/buildtools/blob/73375848b73a586a8ac439a1263b605901414ad8/src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets#L79) in Microsoft.DotNet.Build.CloudTestTasks, which are each described below.
+
+## `Test` target
+
+The `Test` target is used to run or archive tests locally. At a high level, this target performs four steps:
+
+1. Copies the required files for running the tests to the test execution directory.
+2. Generates the test execution script, which was introduced for running the tests in Helix but is also used for running them locally.
```

Comment:
Good idea. I've changed it to "distributed automation systems such as [Helix](https://helix.dot.net/)".